homeassistant/components/spider/climate.py

```diff
@@ -44,6 +44,16 @@ def __init__(self, api, thermostat):
         self.api = api
         self.thermostat = thermostat
 
+    @property
+    def device_info(self):
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
```

Comment:
It's more correct to use `self.thermostat.id` as the device id since that is what that is. `self.unique_id` is the id of the entity. In this case they are the same but it's not true per se.

Reply:
@MartinHjelmare I tried that at the beginning, but ended up with errors;
```
2020-12-10 23:23:31 ERROR (MainThread) [homeassistant.components.climate] Error while setting up spider platform for climate
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 207, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 316, in async_add_entities
    await asyncio.gather(*tasks)
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/entity_platform.py", line 405, in _async_add_entity
    device = device_registry.async_get_or_create(**processed_dev_info)
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/device_registry.py", line 259, in async_get_or_create
    return self._async_update_device(
  File "/srv/homeassistant/lib/python3.8/site-packages/homeassistant/helpers/device_registry.py", line 352, in _async_update_device
    if value is not _UNDEF and not value.issubset(old_value):
AttributeError: 'str' object has no attribute 'issubset'
```

That's why I took a look how other components did it, which is the current code.

Reply:
We still need to use a set.

```suggestion
            "identifiers": {(DOMAIN, self.thermostat.id)},
```

Reply:
All set now :)